Set up Prettier pre-commit hooks

richardzeng1 opened the pull request.

siefkenj commented on package.json:
```diff
@@ -9,7 +9,14 @@
+    "lint-staged": {
+        "*.js": [
+            "prettier --write --tab-width 4",
+            "git add"
```
Does this mean all `.js` files will be committed automatically? If so, we don't want that.

richardzeng1 replied:
No, it runs prettier on .js files staged for commit. Don't merge this branch yet, I'll add pre-commit hooks for .vue and .json

richardzeng1 committed:
Add Pre-commit hook
Add pre-commit hooks for .json, .vue and .js
Added isomorphic-fetch and es6-promise dependency for testing

richardzeng1 committed:
Merge branch 'master' into pre-commit-hook
